blockchain/state_migration.go

```diff
@@ -499,13 +510,25 @@ func (bc *BlockChain) StartWarmUp() error {
 			bc.quitWarmUp = nil
 		}()
 
-		children, err := db.TrieDB().NodeChildren(block.Root())
+		root := block.Root()
+		if !common.EmptyAddress(contractAddr) {
```

Comment:
How about moving this check logic before this goroutine and making this can return an error?

Reply:
Ok, that would be better.